src/common/marlin_print_preview.cpp

```diff
@@ -166,8 +166,8 @@ auto PrintPreview::check_tools_mapping_validity(const ToolMapper &mapper, const
             return true;
         }
 
-        float nozzle_diameter_distance = gcode.get_extruder_info(gcode_tool).nozzle_diameter.value() - get_nozzle_diameter(physical_extruder);
-        if (nozzle_diameter_distance > 0.001f || nozzle_diameter_distance < -0.001f) {
+        float nozzle_diameter_distance = std::abs((float)gcode.get_extruder_info(gcode_tool).nozzle_diameter.value() - (float)get_nozzle_diameter(physical_extruder));
```

Comment:
The overall preference is to use C++-style casts (`static_cast` here) where possible. If you find some C-style casts around, it's mostly just leftovers from ancient times 😇

Reply:
Is resolved :D